### PR TITLE
[pr2eus] Add key topic-name to :send-cmd-vel-raw

### PR DESCRIPTION
**Pull request by nakane11**
Currently `:send-cmd-vel-raw` publishes topic set in `robot-move-base-interface :init`. 
This PR enables specifying topic name when calling the method. It is useful when you want to use subclass such as `pr2-interface`.

Added key `topic-name` follows `:send-cmd-vel-raw` in [fetch-interface](https://github.com/jsk-ros-pkg/jsk_robot/blob/c91004358e0e60bdeca5d05f1aa96e01bf970947/jsk_fetch_robot/fetcheus/fetch-interface.l#L355-L364).

### CODE REVIEW

**Comment by tkmtnt7000** [comment]
Nice PR!!

File: pr2eus/robot-interface.l [hunk 1]
```diff
@@ -1592,17 +1592,23 @@ Return value is a list of interpolatingp for all controllers, so (null (some #'i
...
    (when (send self :simulation-modep)
      (return-from :send-cmd-vel-raw t))
-   (unless (ros::get-topic-publisher (if namespace (format nil "~A/~A" namespace cmd-vel-topic) cmd-vel-topic))
-     (ros::advertise (if namespace (format nil "~A/~A" namespace cmd-vel-topic) cmd-vel-topic) geometry_msgs::Twist 1)
+   (setq vel-topic
+         (if topic-name
+             (if namespace
+                 (if (eq (char "/" 0) (char topic-name 0)) topic-name (format nil "~A/~A" namespace topic-name))
+                 topic-name)
+             (if namespace (format nil "~A/~A" namespace cmd-vel-topic) cmd-vel-topic)))
```
Comment (tkmtnt7000) [inline pr2eus/robot-interface.l:1603]: It may be better to use `let` instead of `setq` to set `vel-topic` as a local variable.

Comment (nakane11) [reply pr2eus/robot-interface.l]: Thanks for the review. I have revised it.

**Comment by Affonso-Gui** [comment]
> It is useful when you want to use subclass such as pr2-interface.

What do you mean? Why don't you just set the correct topic name in the :init function? Or do we have multiple base_controller servers?

**Comment by nakane11** [comment]
I just missed [args](https://github.com/jsk-ros-pkg/jsk_pr2eus/blob/e77d628d85453bc32917152c7ae8a8ef0c9d0b63/pr2eus/pr2-interface.l#L46) in send-super*. I can set the topic in the :init function as you said.

I would like to use :send-cmd-vel-raw with (pr2-init), but to make pr2-interface with topic name is substitutable for it.

File: pr2eus/robot-interface.l [hunk 2]
```diff
@@ -1592,17 +1592,23 @@ Return value is a list of interpolatingp for all controllers, so (null (some #'i
...
    (when (send self :simulation-modep)
      (return-from :send-cmd-vel-raw t))
-   (unless (ros::get-topic-publisher (if namespace (format nil "~A/~A" namespace cmd-vel-topic) cmd-vel-topic))
-     (ros::advertise (if namespace (format nil "~A/~A" namespace cmd-vel-topic) cmd-vel-topic) geometry_msgs::Twist 1)
+   (let ((vel-topic
+         (if topic-name
+             (if namespace
+                 (if (eq (char "/" 0) (char topic-name 0)) topic-name (format nil "~A/~A" namespace topic-name))
+                 topic-name)
+             (if namespace (format nil "~A/~A" namespace cmd-vel-topic) cmd-vel-topic))))
```
Comment (Affonso-Gui) [inline pr2eus/robot-interface.l:1603]:
```lisp
  (:send-cmd-vel-raw
   (x y d &key (topic-name cmd-vel-topic) ((:namespace ns) namespace)) ;; [m] [m] [degree]
    (if ns (setq topic-name (format nil "~A/~A" ns topic-name)))
    ;; ...
```

**Review by Affonso-Gui — COMMENTED** [review]
Another coding suggestion, in case you are fine on setting `:namespace nil` for global topics (the first letter is not checked)

File: pr2eus/robot-interface.l (hunk 2 above)
Comment (nakane11) [reply pr2eus/robot-interface.l]: Thank you for the helpful advice. It's simple and easy to understand.

**Review by Affonso-Gui — APPROVED** [review]

**Review by tkmtnt7000 — APPROVED** [review]

**Pull request merged**